crates/sui-types/src/messages.rs

```diff
@@ -574,7 +582,7 @@ impl<S> TransactionEnvelope<S> {
     /// Get the transaction digest and write it to the cache
     pub fn digest(&self) -> &TransactionDigest {
         self.transaction_digest
-            .get_or_init(|| TransactionDigest::new(sha3_hash(&self.data)))
+            .get_or_init(|| TransactionDigest::new(sha3_hash(&self.signed_data.data)))
```

Comment:
digest should be on signed_data?

Reply:
The whole idea of this PR is to include user's signature as part of the txID (along with data). Iiuc `signed_data` carries both the tx_data and tx_signature, which is the expected payload.

Wondering though if `Signature` carries the pubkey as well, because in theory we only need the actual signature bytes (it doesn't harm to have all of it though)

Reply:
actually yes @lxfind is correct that the calculation of the digest is incorrect and this line needs to change to include the signature as well, otherwise the whole PR is effectively a noop

Reply:
Oh I meant the whole `signed_data` not `signed_data.data`

Reply:
yep, specifically this needs to change to:

```suggestion
            .get_or_init(|| TransactionDigest::new(sha3_hash(&self.signed_data)))
```

Reply:
Sorry for missing the hash traits off, I've made changes to fix these.